File: app/src/main/java/com/huanchengfly/tieba/post/api/models/UserLikeForumBean.java
package com.huanchengfly.tieba.post.api.models;

import com.google.gson.annotations.SerializedName;
import com.huanchengfly.tieba.post.models.BaseBean;

import java.util.List;

public class UserLikeForumBean extends BaseBean {
    @SerializedName("error_code")
    private String errorCode;
    @SerializedName("error_msg")
    private String errorMsg;
    @SerializedName("has_more")
    private String hasMore;
    @SerializedName("forum_list")
    private ForumListBean forumList;
    @SerializedName("common_forum_list")
    private ForumListBean commonForumList;

    public String getErrorCode() {
        return errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public String getHasMore() {
        return hasMore;
    }

    public ForumListBean getForumList() {
        return forumList;
    }

    public ForumListBean getCommonForumList() {
        return commonForumList;
    }

    public static class ForumListBean {
        @SerializedName("non-gconforum")
        private List<ForumBean> forumList;

        public List<ForumBean> getForumList() {
            return forumList;
        }
    }

    public static class ForumBean {
        private String id;
        private String name;
        @SerializedName("level_id")
        private String levelId;
        @SerializedName("favo_type")
        private String favoType;
        @SerializedName("level_name")
        private String levelName;
        @SerializedName("cur_score")
        private String curScore;
        @SerializedName("levelup_score")
        private String levelUpScore;
        private String avatar;
        private String slogan;

        public String getId() {
            return id;
        }

        public String getName() {
            return name;
        }

        public String getLevelId() {
            return levelId;
        }

        public String getFavoType() {
            return favoType;
        }

        public String getLevelName() {
            return levelName;
        }

        public String getCurScore() {
            return curScore;
        }

        public String getLevelUpScore() {
            return levelUpScore;
        }

        public String getAvatar() {
            return avatar;
        }

        public String getSlogan() {
            return slogan;
        }
    }

}

File: app/src/main/java/com/huanchengfly/tieba/post/api/models/UserLikeForumBean.kt
package com.huanchengfly.tieba.post.api.models

import com.google.gson.annotations.SerializedName
import com.huanchengfly.tieba.post.models.BaseBean
import kotlinx.collections.immutable.persistentListOf
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class UserLikeForumBean(
    @SerialName("error_code")
    @SerializedName("error_code")
    val errorCode: String = "-1",

    @SerialName("error_msg")
    @SerializedName("error_msg")
    val errorMsg: String = "unknown error",

    @JvmField
    @SerialName("has_more")
    @SerializedName("has_more")
    val hasMore: String = "0",

    @JvmField
    @SerialName("forum_list")
    @SerializedName("forum_list")
    val forumList: ForumListBean = ForumListBean(),

    @SerialName("common_forum_list")
    @SerializedName("common_forum_list")
    val commonForumList: ForumListBean = ForumListBean(),
) : BaseBean() {

    @Serializable
    data class ForumListBean(
        @JvmField
        @SerialName("non-gconforum")
        @SerializedName("non-gconforum")
        val forumList: List<ForumBean> = persistentListOf(),
    )

    @Serializable
    data class ForumBean(
        val id: String? = null,

        @JvmField
        val name: String? = null,

        @SerialName("level_id")
        @SerializedName("level_id")
        val levelId: String? = null,

        @SerialName("favo_type")
        @SerializedName("favo_type")
        val favoType: String? = null,

        @SerialName("level_name")
        @SerializedName("level_name")
        val levelName: String? = null,

        @SerialName("cur_score")
        @SerializedName("cur_score")
        val curScore: String? = null,

        @SerialName("levelup_score")
        @SerializedName("levelup_score")
        val levelUpScore: String? = null,

        @JvmField
        val avatar: String? = null,

        @JvmField
        val slogan: String? = null,
    )
}
